tests/test_mock.py

```diff
@@ -41,13 +41,14 @@ def test_simple(mock_matplotlib):
     bins = [0, 1, 2, 3]
     hep.histplot(h, bins, yerr=True, label="X", ax=ax)
 
-    assert len(ax.mock_calls) == 12
+    assert len(ax.mock_calls) == 13
 
     ax.stairs.assert_called_once_with(
         values=approx([1.0, 3.0, 2.0]),
         edges=approx([0.0, 1.0, 2.0, 3.0]),
         baseline=0,
         label=None,
+        color=plt.rcParams["axes.prop_cycle"].by_key()["color"][-1],
```

Comment:
Here the color cycle is not resetting to the original order as `ax.has_data()` (which, if false reset the color cycle) here calls `mock.has_data()` which is always true.